src/client-rs/src/client/client.rs

```diff
@@ -12,9 +12,9 @@ use flv_future_aio::net::tls::AllDomainConnector;
 
 use crate::ClientError;
 
-/// Generic client trait
+/// Frame with request and response
 #[async_trait]
-pub trait Client: Sync + Send {
+pub trait SerialFrame: Sync + Send {
```

Comment:
What is the inspiration for this name? Does this trait describe requests? Does it describe a way for building requests?

By looking at the function names, it seems like this trait is pretty broad in what it does, it doesn't really give me a good idea as to what an implementor might look like. What is the core functionality that you're abstracting over? I don't see any instances in the codebase where a trait bound is used (`fn thing<SF: SerialFrame>(frame: SF)`). I read a good blog post a few weeks ago about [Concrete Abstraction](https://matklad.github.io/2020/08/15/concrete-abstraction.html). The idea is basically if you always know the concrete type at your call site, then it works just as well to make two types with matching function signatures, you don't actually need to abstract over them with a trait.

That's probably more review than this renaming required, but if we don't need the trait then we also don't need to name it, right? 😉

Reply:
It is intended to send frame using request and response manner.   But I think this doesn't need to be exposed publicly so we don't have decide name for now.